# Results


## QC

### PhiX
No significant hits were identified to indicate contamination with PhiX

### BUSCO
Busco 4.0.2 was used to estimate the completeness of the genome assembly and annotation. BUSCO in genome mode predicted C:64.4%[S:62.0%,D:2.4%],F:3.1%,M:32.5%,n:255 and C:88.1%[S:86.0%,D:2.1%],F:2.1%,M:9.8%,n:13335 on eukaryota_odb10 and cetartiodactyla_odb10, respectively.  In protein mode, a remarkable score improvement is seen in both eukaryota and cetartiodactyla at C:97.7%[S:41.2%,D:56.5%],F:1.2%,M:1.1%,n:255 and C:92.1%[S:40.2%,D:51.9%],F:1.7%,M:6.2%,n:13335.


## Assembly
Without a genome sequence, identifying Elk genes that respond to Brucella infection would be impossible. Here we present the first pseudomolecule assembly of the rocky mountain elk, generated with xxx of sequencing data and xx fold coverage of the genome. An initial assembly of 23,302 contigs was generated using MaSuRCA with an N50 of xx and L90 of xxx contigs. This assembly was assembled into 35 psuedomolecules using 150 gbp of HiC sequencing and the Juicer, Juice Box, and the 3D-DNA pipeline. Assembly quality was verified via mapping illumina and Pacbio CCS reads to the assembly, with alignment percentages at 90.7% and 87.3% respectively.   


## Mitochondrial Genome

The mitochondrial genome is 16,429 bases long with 39 gene models.

Three large mitochondrial integrations were identified in the genome ranging from 93-99% identical on HiC_scaffold_23, HiC_scaffold_33, and HiC_scaffold_34.


## Gene Prediction
For gene annotation we generated 1.5 billion paired end reads of sequencing from six tissues, including kidney, lung, Mes, muscle, scap, and spleen.

## Repeat prediction
Two repeat predictions were used on the genome, annotating common areas, but also distinct.  With EDTA, DNA transposons comprised the largest percentage of repeats in the genome, at 16%. Both Copia and Gypsy retrotransposons  were present at 1% of the genome, while Helitrons were nearly twice as large at 2% of the genome. While EDTA relies heavily on homology and transposon structure, Repeatmodeler/Repeatmasker uses homology and copy number to determine repeats in the genome.  Repeatmodeler  assessed xx percent of the genome as repetitive, with xx contributing the most to the genome's repetitiveness.

## Functional Annotation
